task_sdk/src/airflow/sdk/api/client.py

```diff
@@ -161,9 +164,19 @@ class ConnectionOperations:
     def __init__(self, client: Client):
         self.client = client
 
-    def get(self, conn_id: str) -> ConnectionResponse:
+    def get(self, conn_id: str) -> ConnectionResponse | ErrorResponse:
         """Get a connection from the API server."""
-        resp = self.client.get(f"connections/{conn_id}")
+        try:
+            resp = self.client.get(f"connections/{conn_id}")
+        except ServerResponseError as e:
+            if e.response.status_code == HTTPStatus.NOT_FOUND:
+                log.error(
+                    "Connection not found",
+                    conn_id=conn_id,
+                    detail=e.detail,
+                    status_code=e.response.status_code,
+                )
+                return ErrorResponse(error=ErrorType.CONNECTION_NOT_FOUND, detail={"conn_id": conn_id})
```

Comment:
Return seems an odd choice to me. Was there a reason you didn't throw?

Reply:
Yea, I needed to pass some message from supervisor to task runner when the API returned 404. 

I had 2 options, either to catch the exception here in the client.py or in `_handle_request` in Supervisor. I opted to do it here, since doing it in handle request didn't feel right -- just because I thought API client should be able to handle the status code and pass the info -- as part of its contract.

Reply:
Open to other ideas though: We just need a way for the connection requestor (in ConnectionAccessor in Task Runner via SUPERVISOR_COMMS) to know that the connection wasn't found. Since, it will need something when it does `SUPERVISOR_COMMS.get_message`.

Reply:
Yeah I agree that the api client is a better place to handle the error than even pushing it down to the supervisor. The supervisor's job is to delegate the API call to the client which will handle it and give it a response/error